src/v2/Apps/ViewingRoom/ViewingRoomApp.tsx

```diff
@@ -27,11 +27,15 @@ const ViewingRoomApp: React.FC<ViewingRoomAppProps> = ({
   const { mediator, user } = useContext(SystemContext)
   useEffect(() => {
     if (!user || !user.id) {
-      openAuthModal(mediator, {
-        mode: ModalType.signup,
-        redirectTo: window.location.href,
-        contextModule: ContextModule.viewingRoom,
-        intent: Intent.viewViewingRoom,
+      // openAuthModal will fire off "open:auth" event before ModalContainer
+      // is mounted, use setImmediate to fire after next tick
+      setImmediate(() => {
+        openAuthModal(mediator, {
+          mode: ModalType.signup,
+          redirectTo: window.location.href,
+          contextModule: ContextModule.viewingRoom,
+          intent: Intent.viewViewingRoom,
+        })
```

Comment:
Not 100% sure but I think the reason the specs are failing is that this needs to be torn down properly.

`useEffect` accepts a return function that runs when the component unmounts.

Try:

```tsx
  useEffect(() => {
    if (user || user.id) return

    // openAuthModal will fire off "open:auth" event before ModalContainer
    // is mounted, use setImmediate to fire after next tick
    const immediateID = setImmediate(() => {
      openAuthModal(mediator, {
        mode: ModalType.signup,
        redirectTo: window.location.href,
        contextModule: ContextModule.viewingRoom,
        intent: Intent.viewViewingRoom,
      })
    })

    return () => clearImmediate(immediateID)
  }, [user, mediator])
```

Reply:
🙏 thanks! i was messing around and ended up getting tests to pass with `setTimeout((), 0)`, which is weird. i'll make the changes you suggested.

Reply:
Ah cool, if you choose to use the `setTimeout` you should include the teardown, regardless — just using `clearTimeout` instead of `clearImmediate`.